### PR TITLE
Refactor/threading

### PR DESCRIPTION
**Pull request by ailox**

### CODE REVIEW

**Comment by bynflow** [comment]
Refactor_threading_performance_improve

**Comment by ailox** [comment]
Thanks for merging :+1:

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+__pycache__/
+venv/
```
Comment (bynflow) [inline .gitignore:2]: Is this neccessary?

Comment (ailox) [reply .gitignore:2]: This was only required on my system, but it should be useful for anyone that wants to work on this project.

**Pull request merged**